### ISSUE TITLE
[Enhancement] Spot illustrations

### ISSUE THREAD

**Issue by henrikvoetmand** [kirbydesign/designsystem#3426]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
To further delight the products we are introducing a set of larger icons with more colors and detail.
In order to support multiple brands we need a solid and dynamic way to change the icons colors.
The icons varies in detail and design in relation to their size. We need way to show different icon svg's depending on the size.

## Describe the solution you'd like
**UX provides:**
- 3 svg files for each icon variant with seperated paths for areas that need color change

**Kirby or DRB provides:**
See summary of tech refinement below...

## Are there any additional context?
![Icons - example](https://github.com/kirbydesign/designsystem/assets/71710012/351697ce-7662-4845-ab31-eda6b9591495)

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-refinement)
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by RasmusKjeldgaard** [comment]
Summary of tech refinement with @jakobe 

### Concept
While spot illustrations are conceptually similar to our existing icons, the nitty gritties differ: 

1. Spot illustrations have no relation to the existing icons. Any existing icon should still have its current implementation of simple scaling of the svg for different sizes.
   - E.g. the calendar spot illustration is a different concept with another default svg
2. Spot illustrations do not have an `xs` size, and actual size of `sm`, `md`, `lg` is different from current `kirby-icon` size.
3. Multiple svgs exist per spot illustration, as each size can have small variations, detail levels etc.

Given these differences, and our slightly over-complicated implementation of kirby-icon, we think it is best to implement this as a new component using the ion-icon component.

With `ion-icon`, we get functionality for fetching the svg from source and inlining it, making it styleable. In addition, there are some performance benefits as icons are not loaded before they become visible to the user. 

### Tech refinement
#### :accessibility: Accessibility 
The spot illustration is purely decorative, so it should be omitted from assistive technology like screen-readers etc.
As `ion-icon` renders with `role="img"` both `ion-icon` and the inlined `svg` will render as an image in the a11y tree, so `aria-hidden` should be set on the component itself.

#### Colors
ion-icon handles loading the svg from source, which means that the icon svg is inlined inside the ion-icon shadow root. 
This means we should add part attributes to be able to style the individual paths from the outside. 
The branded details and colored background of the svgs could be implemented something like this:
```
<svg xmlns="http://www.w3.org/2000/svg" width="97" height="84" viewBox="0 0 97 84">
    <g fill="none" fill-rule="evenodd">
        <path part="background" d="..."/>
        ...
        <path part="detail" d="..."/>
    </g>
</svg>
```

This means we can target the details and background for any icon from within the spot illustrations component styles like this, with a sensible default that still allows consumers to overwrite on the application level or locally per component via `--kirby-spot-illustration-detail` and `--kirby-spot-illustration-background`.

```
ion-icon::part(detail) {
  fill: var(--kirby-spot-illustration-detail, var(--kirby-primary));
}

ion-icon::part(background) {
  fill: var(--kirby-spot-illustration-background, var(--kirby-semi-light));
}
```

##### Inside Card
The default background fill is `kirby-semi-light` when the spot illustration is used on canvas.
When the spot illustration is used inside a (white) card the background fill should be  `kirby-light`.


#### Multiple svgs
We should provide a way to register multiple svgs under a specific spot illustration name, that is automatically selected based on the given size.

Some fallback logic should be in place if a specific icon for the given size is not registered - possibly just a 'base' svg that is naively scaled across sizes that have no corresponding svg?

It should be possible to extend the [IconRegistryService](https://github.com/kirbydesign/designsystem/blob/develop/libs/designsystem/icon/src/icon-registry.service.ts) with functionality for getting icons based on a specific size, given a structure like this:

```
{
    name: 'calendar',
    base: 'assets/spotillustrations/calendar-base.svg',
    sizes: {
        sm: 'assets/spotillustrations/calendar-sm.svg',
        md: 'assets/spotillustrations/calendar-md.svg',
        lg: 'assets/spotillustrations/calendar-lg.svg',
    },
},
```

#### Assets package
Assets could for now just be svgs in a sub-folder. When new svgs are added we will have to publish a new package, but we should be able to publish rather smoothly.

Consumers will have to copy them [at build time similarly to the existing icons](https://github.com/kirbydesign/designsystem?tab=readme-ov-file#icons).

**Comment by RasmusKjeldgaard** [comment]
Status update:

Based on implementation and clarifications by @troelslenda in #3996, a couple of questions need to be settled:

- Should we prepare spot illustrations for different themes, e.g. dark mode or brand color backgrounds?
  - If so, the SVGs should probably be revisited and reimplemented with cutouts instead of colored surfaces with bg color where elements overlap as this gives additional implementation burdens that can be avioded with transparent areas instead.
- Should Spot illustrations always have designs in the four variants?
  - Or are we ok with scaling them 'naively' in some cases?
  - If we want to scale them naively, can we implement with stroke and hidden/visible elements, so there's only one SVG?
- Are the base icons just regular kirby icons? Do they exist in the spot illustrations concept?
  - Why does small size exist?

**Comment by RasmusKjeldgaard** [comment]
@henrikvoetmand was able to clarify the questions above.

- We want to prepare the SVGs for themes, so they should be revisited and made with cutouts etc.
  - This is done by the designers, but preferably with input from @troelslenda on the specifics for the first couple of illustrations
- The base icons are not part of the concept, they are just in the spec for reference.
- Without base size, there is still four variants as an 'xl' size now exists.
  - The sizes/variants are: sm, md, lg, xl
- The designers will deliver an svg for each size, so we never have to scale icons up/down
- We think it is best to create the variants as separate svg's so that the concept supports custom design for each icon, even though it might not be necessary just yet with the existing illustrations. 
- There is a naming convention for the icons as outlined in the [zeplin guidelines](https://app.zeplin.io/project/60337f8fa4b49814fda1b48a/screen/662102c06ed90ad595acbb67).
  - Designers will deliver four files following that naming structure

**Comment by troelslenda** [comment]
When looking at the specification example of background

<img width="707" height="179" alt="Image" src="https://github.com/user-attachments/assets/6c0043fe-005f-4f2f-b797-441344056db2" />

There's an indication that what was previously considered "cutaway/cutout" is actually a visible element (white) when the background is light gray (left most example)

I see two options. 
1. Revising the SVG's as previously proposed, but keeping the path and adjusting opacity to 0 when needed.
2. Leaving SVG's as-is and adjusting the fill color of that element when needed on colored backgrounds. (Then the SVG, or it's styling needs to be aware of that specific color - not ideal)

Anyway, as part of the standards, I would like to see a rule saying that there can be no overlapping paths, as this would make it much easier to choose option one.

**Comment by henrikvoetmand** [comment]
There was an error in the example.

> Illustrations inherit the fill color from its background.

As described in the guidelines spec :-)

So "Cutouts" should be the approach.

**Comment by troelslenda** [comment]
In light of #3704 and #3705, I'm beginning to question whether spot illustrations truly belong in the kirby-extension package. Wouldn't placing them there introduce a dependency from kirby to its own extension, thereby breaking our architectural conventions?
This seems especially relevant if `alert` and `message state` are intended to remain simple and non-modular. While we could enable content projection to support `spot illustration` in these components, that doesn’t seem like a clean solution either. 

That said, I consider the PR complete from a code perspective — this might just be a design consideration to revisit later.

**Comment by troelslenda** [comment]
I've also instructed the designers working on upcoming spot illustrations to follow the ruleset I added to Storybook, to ensure consistency and alignment going forward.
